platforms/core-configuration/model-core/src/test/groovy/org/gradle/api/internal/provider/MapPropertySpec.groovy

```diff
@@ -1326,6 +1327,37 @@ The value of this property is derived from: <source>""")
         property.explicit
     }
 
+    def "'#label' vs undefined-safety"() {
```

Comment:
I intend to make this test case mimic `CollectionPropertySpec's` more closely (and get rid of the corresponding tests as methods).